### PR TITLE
DA_DP_TC069_Verify that user is unable to proceed to next step or finish creating data profile if  "Name *" field is left empty

### PR DESCRIPTION
**Pull request by taile47711**

### CODE REVIEW

File: TADashboard/src/main/java/page_objects/DashboardPage.java [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
     private Label lblWelcome = new Label("css=a[href='#Welcome']");
+    private Link lnkDataProfiles = new Link("css=li>ul>li>a[href='profiles.jsp']");
+    private Link lnkAdminister = new Link("css=a[href='#Administer']");
 
     public boolean doesWelcomeLabelDisplay() {
         return lblWelcome.isVisible();
     }
-}
+
+    public void clickLnkDataProfiles(){
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/DashboardPage.java]: you should have a method like `selectAdministerMenu` then the usage is `selectAdministerMenu("Data Profiles")` or `selectAdministerMenu("Panel")`

File: TADashboard/src/main/java/page_objects/DataProfilesPage.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package page_objects;
+
+import com.logigear.control.common.imp.Link;
+
+public class DataProfilesPage {
+    private Link lnkAddNew = new Link("css=div[class='panel_tag2'] a[href='profile.jsp?action=create']");
+
+    public void clickLnkAddNew(){
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/DataProfilesPage.java]: again, I'm not sure why you guys keep having this naming issue, why don't you just name it `clickAddNew` or `clickAddNewLink`?

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+    public void clickBtnNext() {
+        btnNext.click();
+    }
+
+    public void clickBtnFinish() {
+        btnFinish.click();
+    }
+
+    public String isAlertVisible() {
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: what is this? if you need something like getting alert text in DriverUtils, please let me or the team leader, we will have that added

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        return DriverUtils.getWebDriver().switchTo().alert().getText();
+    }
+
+    public void clickOKInAlert() {
+        DriverUtils.acceptAlert();
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: why do you need a new method for accepting alert?

File: TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import utils.common.Constants;
+import utils.helper.Logger;
+
+public class NegativeGeneralSettingsTest extends BaseTest {
+    GeneralSettingsPage generalSettingsPage = new GeneralSettingsPage();
+    LoginPage loginPage = new LoginPage();
+    DataProfilesPage dataProfilesPage = new DataProfilesPage();
+    DashboardPage dashboardPage = new DashboardPage();
+
+    @Test(description = "Verify that user is unable to proceed to next step or finish creating data profile if  \"Name *\" field is left empty")
```
Comment (tanmle) [inline TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java]: why don't you use single quote in the "Name"?

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java (hunk 3 above)
Comment (tanmle) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: and the method is wrong, it is a String return, why do you name it `is...`?

Comment (MinhQuan119) [inline TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: it should be getAlertText() instead
only use `is...()` or `does....()` method if it returns boolean

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+public class GeneralSettingsPage {
+    private Button btnNext = new Button("css=input[value='Next']");
+    private Button btnFinish = new Button("css=input[value='Finish']");
+
+    public void clickBtnNext() {
+        btnNext.click();
+    }
+
+    public void clickBtnFinish() {
```
Comment (MinhQuan119) [inline TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: It should be clickFinishButton()
Btn, lbl, lnk, etc only use for Locators and Elements, do not use it for Methods
Please apply that for the rest

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java (hunk 3 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: I fixed it. Please check again.

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java (hunk 6 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: I fixed it. Please check again.

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java (hunk 3 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: I fixed this issue 2 hours ago

File: TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java (hunk 5 above)
Comment (taile47711) [reply TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java]: I fixed it. Please check again.

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java (hunk 4 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: I fixed it. Please check again.

File: TADashboard/src/main/java/page_objects/DashboardPage.java (hunk 1 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/DashboardPage.java]: I fixed it. Please check again.

File: TADashboard/src/main/java/page_objects/DataProfilesPage.java (hunk 2 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/DataProfilesPage.java]: I fixed it. Please check again.

File: TADashboard/src/main/java/page_objects/DashboardPage.java (hunk 1 above)
Comment (MinhQuan119) [reply TADashboard/src/main/java/page_objects/DashboardPage.java]: It wasn't fixed like the way Tan said. Please recheck his comment

File: TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java (hunk 5 above)
Comment (MinhQuan119) [reply TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java]: It was not fixed that way, I don't see any single quote here.

File: TADashboard/src/main/java/page_objects/DashboardPage.java (hunk 1 above)
Comment (MinhQuan119) [reply TADashboard/src/main/java/page_objects/DashboardPage.java]: It was not fixed like he said. You need to read his comment again

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java (hunk 3 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: I fixed it. Please check again

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+    public void clickNextButton() {
+        btnNext.click();
+    }
+
+    public void clickFinishButton() {
+        btnFinish.click();
+    }
+
+    public String getAlertText() {
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: you don't need this to wrap the getAlertText()

File: TADashboard/src/main/java/page_objects/DashboardPage.java [hunk 8]
```diff
@@ -1,11 +1,19 @@
...
     private Label lblWelcome = new Label("css=a[href='#Welcome']");
+    private Link lnkDataProfiles = new Link("css=li>ul>li>a[href='profiles.jsp']");
+    private Link lnkAdminister = new Link("css=a[href='#Administer']");
 
     public boolean doesWelcomeLabelDisplay() {
         return lblWelcome.isVisible();
     }
-}
+
+    public void selectAdministerMenu(DataProfilesPage dataProfilesPage){
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/DashboardPage.java]: what is this? what is DataProfilesPage, and where you are using it in this method?

**Review by tanmle — CHANGES_REQUESTED** [review]

**Comment by taile47711** [comment]
image

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+    public void clickNextButton() {
+        btnNext.click();
+    }
+
+    public void clickFinishButton() {
+        btnFinish.click();
+    }
+
+    public void selectAdministerMenu(DataProfilesPage dataProfilesPage) {
```
Comment (MinhQuan119) [inline TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: > you should have a method like selectAdministerMenu then the usage is selectAdministerMenu("Data Profiles") or selectAdministerMenu("Panel")

Do you understand this?

Comment (taile47711) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: I fixed it. Please check again

File: TADashboard/src/main/java/page_objects/DataProfilesPage.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+public class DataProfilesPage {
+    private Link lnkAddNew = new Link("css=div[class='panel_tag2'] a[href='profile.jsp?action=create']");
+    private Link dynamicAdministerMenu = new Link("xpath=//ul[@id='ulAdminister']/li/a[text()='%s']");
+    private Link lnkAdminister = new Link("css=a[href='#Administer']");
+
+    public void clickAddNewLink(){
+        lnkAddNew.click();
+    }
+
+    public void selectAdministerMenu(String pageName) {
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/DataProfilesPage.java]: Administer menu is common, it is not just for DataProfiles

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java [hunk 11]
```diff
@@ -0,0 +1,18 @@
+package page_objects;
+
+import com.logigear.control.common.imp.Button;
+import com.logigear.control.common.imp.Link;
+import com.logigear.driver.DriverUtils;
+
+public class GeneralSettingsPage {
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: can you show me GeneralSetting page in the UI?

File: TADashboard/src/test/resources/suites/test.xml [hunk 12]
```diff
@@ -8,7 +8,13 @@
     <test name="Run Chrome Local">
         <parameter name="Browser" value="chrome.local" />
         <classes>
-            <class name="tests.login.PositiveLoginTests"></class>
+            <class name="tests.general_settings.NegativeGeneralSettingsTest"></class>
+        </classes>
+    </test>
+    <test name="Run Firefox Local">
```
Comment (tanmle) [inline TADashboard/src/test/resources/suites/test.xml]: create another suite for data profile.

**Review by tanmle — CHANGES_REQUESTED** [review]

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java (hunk 11 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: image

File: TADashboard/src/test/resources/suites/test.xml (hunk 12 above)
Comment (taile47711) [reply TADashboard/src/test/resources/suites/test.xml]: I fixed it. Please check again

File: TADashboard/src/main/java/page_objects/DataProfilesPage.java (hunk 10 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/DataProfilesPage.java]: Please check again.

File: TADashboard/src/main/java/page_objects/GeneralSettingsPage.java (hunk 11 above)
Comment (tanmle) [reply TADashboard/src/main/java/page_objects/GeneralSettingsPage.java]: this is DataProfilesGeneralSettingPage

File: TADashboard/src/main/java/page_objects/BasePage.java [hunk 13]
```diff
@@ -0,0 +1,14 @@
+package page_objects;
+
+import com.logigear.control.common.imp.Link;
+
+public class BasePage {
+    private Link dynamicNavigateToPage = new Link("xpath=//ul[@id='ulAdminister']/li/a[text()='%s']");
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/BasePage.java]: just name is dynamicAdministerMenu

File: TADashboard/src/main/java/page_objects/DataProfilesPage.java [hunk 14]
```diff
@@ -0,0 +1,11 @@
+package page_objects;
+
+import com.logigear.control.common.imp.Link;
+
+public class DataProfilesPage extends BasePage{
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/DataProfilesPage.java]: code format

**Review by tanmle — CHANGES_REQUESTED** [review]

File: TADashboard/src/main/java/page_objects/BasePage.java (hunk 13 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/BasePage.java]: check again

File: TADashboard/src/main/java/page_objects/DataProfilesPage.java (hunk 14 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/DataProfilesPage.java]: Please check again

File: TADashboard/src/test/resources/suites/test.xml (hunk 12 above)
Comment (taile47711) [reply TADashboard/src/test/resources/suites/test.xml]: Please check agian

File: TADashboard/src/main/java/page_objects/BasePage.java [hunk 15]
```diff
@@ -0,0 +1,14 @@
+package page_objects;
+
+import com.logigear.control.common.imp.Link;
+
+public class BasePage {
+    private Link dynamicAdministerMenu = new Link("xpath=//ul[@id='ulAdminister']/li/a[text()='%s']");
+    private Link lnkAdminister = new Link("css=a[href='#Administer']");
+
+    public void selectAdministerMenu(String pageName) {
```
Comment (tanmle) [inline TADashboard/src/main/java/page_objects/BasePage.java]: menu or pageName?

**Review by tanmle — CHANGES_REQUESTED** [review]

File: TADashboard/src/main/java/page_objects/BasePage.java (hunk 15 above)
Comment (taile47711) [reply TADashboard/src/main/java/page_objects/BasePage.java]: I fixed it. Please check again

**Comment by taile47711** [comment]
image

**Comment by taile47711** [comment]
image

File: TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+        Logger.step("Navigate to data profile page");
+        dataProfilesPage.selectAdministerMenu("Data Profiles");
+
+        Logger.step("Click Add New link");
+        dataProfilesPage.clickAddNewLink();
+
+        Logger.step("Click Next button");
+        dataProfilesGeneralSettingsPage.clickNextButton();
+
+        Logger.verify("Verify that dialog message is display");
```
Comment (tanmle) [inline TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java]: wrong English grammar.

**Review by tanmle — CHANGES_REQUESTED** [review]

File: TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java (hunk 16 above)
Comment (taile47711) [reply TADashboard/src/test/java/tests/general_settings/NegativeGeneralSettingsTest.java]: I fixed it. Please check again

**Review by tanmle — APPROVED** [review]

**Pull request merged**